File: c-algorithm/src/compare-pointer.c
// 识别到的包含指令
#include "compare-pointer.h"

// 识别到的函数定义
// 函数: pointer_equal (line 25)
int pointer_equal(void *location1, void *location2)
{
	return location1 == location2;
}

// 函数: pointer_compare (line 30)
int pointer_compare(void *location1, void *location2)
{
	if (location1 < location2) {
		return -1;
	} else if (location1 > location2) {
		return 1;
	} else {
		return 0;
	}
}

File: comparisons/MinsC2Rust/c_algorithm/Output/test_project/src/compare_pointer.rs

pub fn pointer_equal<T>(location1: &T, location2: &T) -> bool {
    std::ptr::eq(location1, location2)
}

pub fn pointer_compare<T>(location1: &T, location2: &T) -> i32 {
    if location1 as *const _ < location2 as *const _ {
        -1
    } else if location1 as *const _ > location2 as *const _ {
        1
    } else {
        0
    }
}

